src/main/java/cafe/cryptography/curve25519/RistrettoElement.java

```diff
@@ -69,11 +69,17 @@ public CompressedRistretto compress() {
 
     /**
      * Constant-time equality check.
+     * <p>
+     * This is the ristretto255 EQUALS function.
      *
      * @return 1 if this and other are equal, 0 otherwise.
      */
     public int ctEquals(final RistrettoElement other) {
-        throw new UnsupportedOperationException();
+        FieldElement X1Y2 = this.repr.X.multiply(other.repr.Y);
+        FieldElement Y1X2 = this.repr.Y.multiply(other.repr.X);
+        FieldElement Y1Y2 = this.repr.Y.multiply(other.repr.Y);
+        FieldElement X1X2 = this.repr.X.multiply(other.repr.X);
+        return X1Y2.ctEquals(Y1X2) | Y1Y2.ctEquals(X1X2);
```

Comment:
Ask @hdevalence about the intended semantics of `||` and `==` in the spec.

Reply:
Not sure what the question is here, but `||` is supposed to be logical OR and `==` is supposed to be equality

Reply:
The question is whether they are meant to be constant-time or not. I interpreted them as so, but the spec does define `CT_EQ` explicitly, and elsewhere in the spec `|` is used for constant-time OR instead of `||`.

Reply:
Oh! I see. Yes, those should be constant-time ORs, good catch!